wire-library/wire-runtime-swift/src/main/swift/wellknowntypes+Custom/Duration+Codable.swift

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2023 Block Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+#if !WIRE_REMOVE_CODABLE
+
+extension Wire.Duration : Codable {
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.singleValueContainer()
+
+        let encoded: String = {
+            if nanos == 0 {
```

Comment:
This implementation was cribbed from [the Kotlin implementation](https://github.com/square/wire/blob/master/wire-library/wire-runtime/src/jvmMain/kotlin/com/squareup/wire/internal/DurationJsonFormatter.kt#L48)